Permit '=' separator and '[ipv6]' in --add-host (#4663)
docker/cli: docs/reference/commandline/run.md

round-trip min/avg/max = 92.209/92.495/93.052 ms
```

> IPv6 addresses may be enclosed in square brackets:

Reviewer: ```suggestion
You can wrap an IPv6 address in square brackets:
```

Author: Done.

$ python3 -m http.server 8000
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
$ docker run \
  --add-host host.docker.internal=host-gateway \
  curlimages/curl -s host.docker.internal:8000/hello
hello from host!
```

> The `--add-host` flag will also accept a `:` separator, for example:

Reviewer: ```suggestion
The `--add-host` flag also accepts a `:` separator, for example:
```

Author: Done.

```console

> $ docker run --add-host=docker:93.184.216.34 --rm -it alpine

Reviewer 2: For examples, I try to use names that help identify they're "custom" values (and for illustrative purpose). Using `docker` (or `hostname` as was used elsewhere) _could_ be confused for "docker" being some special value.

Something like `my-hostname` could perhaps work;

```bash
$ docker run --add-host=my-hostname:93.184.216.34 --rm -it alpine
```

I'm curious where the `93.184.216.34` address came from; it 

It's also good to make an example "do" something. In this case, we want to illustrate that adding the hostname allows the hostname to be resolved inside the container. I'm curious where the `93.184.216.34` IP-address came from, but if we want to be sure `ping` works, perhaps we should use a well-known IP-address, e.g. `8.8.8.8`;

```bash
$ docker run --add-host=my-hostname:8.8.8.8 --rm alpine ping -c1 my-hostname
PING my-hostname (8.8.8.8): 56 data bytes
64 bytes from 8.8.8.8: seq=0 ttl=63 time=14.995 ms

--- my-hostname ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 14.995/14.995/14.995 ms
```

The advantage of `ping` of course is that (most) users would understand what we're doing. Alternatively (if we don't want to depend on networking, and only illustrate the actual effect), we could show that the entry was added to `/etc/hosts` inside the container (but not sure if that's to be considered too much an implementation details);

```bash
$ docker run --add-host=my-hostname:8.8.8.8 --rm alpine sh -c 'cat /etc/hosts | grep my-hostname'
8.8.8.8	my-hostname
````

I considered `nslookup` as alternative, but :see_no_evil: of course that doesn't work with `/etc/hosts`, so we'd need something like `getent`, which probably isn't ideal (not as widely used; at least I had to quickly lookup the right way to use it :joy:)

```bash
$ docker run --add-host=my-hostname:8.8.8.8 --rm alpine getent ahosts my-hostname
8.8.8.8         STREAM my-hostname
8.8.8.8         DGRAM  my-hostname
```

Reviewer 2: Oh! Before I forget (and probably not needed for these examples); if we use example _domains_ in our docs; always make sure to use one of the designated domains for that purpose, such as `*.example.com`, `*.example.org` (https://www.rfc-editor.org/rfc/rfc6761.html).

(We've had cases where we used some (assumed) made-up domain, which... was in active use 😬)

Author: `docker:93.184.216.34` was already used in the ping example (the address was introduced in https://github.com/docker/cli/commit/47ba76afb159273e35326bd0cb548e960c51fbc7 ) ... so I kept it, but something like `my-hostname:8.8.8.8` looks better, I'll change it.

"ping" seems like a nice simple/clear example, and with `8.8.8.8` it should normally work, so I'll keep it.
